wgpu-core/src/device/resource.rs

```diff
@@ -1391,9 +1394,15 @@ impl<A: HalApi> Device<A> {
         guard
             .iter(self_id.backend())
             .find(|&(_, bgl)| {
-                bgl.device_id.value.0 == self_id
-                    && bgl.compatible_layout.is_none()
-                    && bgl.entries == *entry_map
+                if bgl.device_id.value.0 != self_id {
+                    return false;
+                }
+
+                if let Some(inner) = bgl.as_inner() {
+                    return inner.entries == *entry_map;
+                }
+
+                false
```

Comment:
nitpick: This diverges less from previous code, and is still just as clear (IMO):

```suggestion
                bgl.device_id.value.0 == self_id
                    && bgl
                        .as_inner()
                        .map_or(false, |inner| inner.entries == *entry_map)
```